pkg/deployment/inventory.go

```diff
@@ -71,15 +71,13 @@ func GenerateRoleInventory(ctx context.Context, helper *helper.Helper,
 		}
 	}
 
-	if instance.Spec.NodeTemplate.Nova != nil {
-		if _, ok := roleNameGroup.Vars["edpm_ovn_metadata_agent_metadata_agent_DEFAULT_nova_metadata_host"]; !ok {
-			novaSvc := &corev1.Service{}
-			err = helper.GetClient().Get(ctx, types.NamespacedName{Name: "nova-metadata-internal", Namespace: instance.Namespace}, novaSvc)
-			if err != nil {
-				return "", err
-			}
-			roleNameGroup.Vars["edpm_ovn_metadata_agent_metadata_agent_DEFAULT_nova_metadata_host"] = novaSvc.Status.LoadBalancer.Ingress[0].IP
+	if _, ok := roleNameGroup.Vars["edpm_ovn_metadata_agent_metadata_agent_DEFAULT_nova_metadata_host"]; !ok {
+		novaSvc := &corev1.Service{}
+		err = helper.GetClient().Get(ctx, types.NamespacedName{Name: "nova-metadata-internal", Namespace: instance.Namespace}, novaSvc)
+		if err != nil {
+			return "", err
```

Comment:
it is being executed by default now, so it will error until it finds `nova-metadata-internal` service.
This breaks the kuttl tests, as the status isn't the expected.
For workaround this, we can add `edpm_ovn_metadata_agent_metadata_agent_DEFAULT_nova_metadata_host` on the kuttl CRs or increase kuttl timeout

Reply:
but we need to check if the nova service is in place, otherwise the reconcile loop would get stuck here when people deploy without the nova service

Reply:
I'll take a look at the kuttl tests.

For the nova service, this is relying on the nova metadata from the control plane. Which I think we can always assume is deployed? Even if nova is not being deployed on the EDPM node, it's no harm to set this variable.

Reply:
on the adoption workflow we enable each service as we adopt it, and we don't have nova adoption yet, so we deploy the ctlplane without it

Reply:
Alright, but the code I'm changing here wasn't checking for nova on the ctlplane either. It was checking if nova-compute was on the EDPM node. So, I think we can change the logic here to just be if the helper.GetClient().Get does not return anything, instead of requeue'ing we just don't set the ansible var.

Reply:
+1

Reply:
I've made that change in: https://github.com/openstack-k8s-operators/dataplane-operator/pull/342